### PR TITLE
Add date minute parameter

### PR DESCRIPTION
**Pull request by themalkolm**
Surprisingly luigi could be used to run pipelines with minute precision in configuration. This represents an extra parameter we already have for our project. Why not share, right?

### CODE REVIEW

**Comment by erikbern** [comment]
Nice!

**Comment by Tarrasch** [comment]
Cool. You've used this in production?

**Comment by themalkolm** [comment]
@Tarrasch, well, we actually use `%Y-%m-%dT%HH%MM` (note an extra M on the end) and `'%Y-%m-%dT%HH%MM%SS`.

File: luigi/parameter.py [hunk 1]
```diff
@@ -309,20 +309,34 @@ class DateHourParameter(Parameter):
...
+        if dt is None:
+            return str(dt)
+        return dt.strftime(self.date_format)
+
+
+class DateMinuteParameter(DateHourParameter):
+    """Parameter whose value is a :py:class:`~datetime.datetime` specified to the minute.
+
+    A DateMinuteParameter is a `ISO 8601 <http://en.wikipedia.org/wiki/ISO_8601>`_ formatted
+    date and time specified to the minute. For example, ``2013-07-10T19H42`` specifies July 10, 2013 at
```
Comment (Tarrasch) [inline luigi/parameter.py]: Please include also (or only) an example with a leading 0 minute. like 19:07 rather than 19:42. How will it be formatted too?

**Comment by Tarrasch** [comment]
Ah wait a minute then, is it ever ever ever a good idea to not include the 0 in the minute? will files look like

``
8
9
10
11
12
```

or 

``
08
09
10
11
12
```

??

**Comment by Tarrasch** [comment]
> Ah wait a minute then

Pun not intended :p

File: luigi/parameter.py (hunk 1 above)
Comment (themalkolm) [inline luigi/parameter.py]: To give a hint that minutes will always have a leading zero if below 10?

**Comment by themalkolm** [comment]
@Tarrasch 

https://docs.python.org/2/library/datetime.html
> %M	Minute as a zero-padded decimal number.	00, 01, ..., 59

**Comment by themalkolm** [comment]
So minutes are always zero-padded.

**Comment by erikbern** [comment]
Merging this, we can always add more tests later

**Comment by Tarrasch** [comment]
> So minutes are always zero-padded.

Awesome then! :)

**Comment by fphilipe** [comment]
Quick comment about the format. I think it is incorrect. I've never seen `H` being used as a separator between the hours and the minutes. I believe this should be a colon instead. Please correct me if I'm wrong.

**Comment by erikbern** [comment]
Yes – you are right. ISO 8601 specifies T between date and time, but between hour and minute it should be colon. `2015-07-28T19:19:34Z` etc

Can we fix this? Sorry I glanced at this PR too quickly

**Comment by fphilipe** [comment]
I suppose the incorrect format would need to be deprecated and supported for a while. I'd be happy to work on a PR.

**Comment by erikbern** [comment]
Another problem is also that colons generally don't work in filenames

**Comment by fphilipe** [comment]
@erikbern ISO 8601 states that times don't necessarily need colons, they can simply be omitted. https://en.wikipedia.org/wiki/ISO_8601#Times

**Comment by erikbern** [comment]
Perfect then let's use 2015-07-29T1356

**Pull request merged**